scrubadub/detectors/spacy_expand_person_title.py

```diff
@@ -0,0 +1,314 @@
+import os
+import re
+import copy
+import logging
+import importlib
+
+from wasabi import msg
+from typing import Generator, Iterable, Optional, Sequence, List
+
+try:
+    import spacy
+    # new in spacy v3
+    from spacy.language import Language
+    from spacy.tokens import Doc
+    from spacy.tokens import Span
+except ImportError as e:
+    if e.name == "spacy":
+        raise ImportError(
+            "Could not find module 'spacy'. If you want to use extras,"
+            " make sure you install scrubadub with 'pip install scrubadub[spacy]'"
+        )
+
+from . import register_detector
+from ..filth import Filth, NameFilth, OrganizationFilth, LocationFilth
+from ..utils import CanonicalStringSet
+from .spacy import SpacyEntityDetector
+
+
+class SpacyExpandPersonTitle(SpacyEntityDetector):
+    """Use spaCy's named entity recognition to identify possible ``Filth``.
+
+    This detector is made to work with v3 of spaCy, since the NER model has been significantly improved in this
+    version.
+
+    This is particularly useful to remove names from text, but can also be used to remove any entity that is
+    recognised by spaCy. A full list of entities that spacy supports can be found here:
+    `<https://spacy.io/api/annotation#named-entities>`_.
+
+    Additional entities can be added like so:
+
+    >>> import scrubadub, scrubadub.detectors.spacy
+    >>> class MoneyFilth(scrubadub.filth.Filth):
+    ...     type = 'money'
+    >>> scrubadub.detectors.spacy.SpacyEntityDetector.filth_cls_map['MONEY'] = MoneyFilth
+    >>> detector = scrubadub.detectors.spacy.SpacyEntityDetector(named_entities=['MONEY'])
+    >>> scrubber = scrubadub.Scrubber(detector_list=[detector])
+    >>> scrubber.clean("You owe me 12 dollars man!")
+    'You owe me {{MONEY}} man!'
+
+    The dictonary ``scrubadub.detectors.spacy.SpacyEntityDetector.filth_cls_map`` is used to map between the spaCy
+    named entity label and the type of scrubadub ``Filth``, while the ``named_entities`` argument sets which named
+    entities are considered ``Filth`` by the ``SpacyEntityDetector``.
+    """
+    filth_cls_map = {
```

Comment:
Much of the code below is identical to that in the spacy detector, since you're inheriting there's no need to have the code written out again, it'll just pick it up from the class it derives from.

Reply:
Please remove all duplicated code.